### PR TITLE
WebRTC Multiplayer peer, documentation

### LINKED ISSUE #12108: WebRTC integration
Hey guys,
Love the work you've all put into this project and don't mean to be a bother but would LOVE to have WebRTC integration for this! Especially with something like PeerJS where all that's needed is a simple identifier for each peer.
Again, thanks for all the work you guys have been doing.

### LINKED ISSUE #14627: High level multiplayer for WebAssembly platform
Will Godot 3.1 support WebAssembly high networked multiplayer?

### PR DESCRIPTION
**Pull request by Faless**
In this PR:

- Add `WebRTCMultiplayer`  (high level multiplayer implementation)
- Normalize `WebRTCDataChannel` properties across browsers.
- Add documentation for WebRTC classes.

The multiplayer implementation is a full p2p mesh, and support a "server emulation mode" where it expects to connect to a peer with id `1` and emits the server connection/disconnection events (see docs for more info).

There is a fork of the multiplayer bomber demo using WebRTC here:
https://github.com/Faless/bomber-rtc

You can try a [live version here](https://no-war.fales.me)

Signaling server is the one from:
https://github.com/Faless/gd-webrtc-signalling

Running the game on native platforms requires usual WebRTC GDNative plugin -> https://github.com/godotengine/webrtc-native/releases/tag/0.3

Closes #14627, closes #12108

### CODE REVIEW

File: modules/webrtc/doc_classes/WebRTCDataChannel.xml [hunk 1]
```diff
@@ -11,85 +11,106 @@
...
 			<description>
+				Returns the maxPacketLifeTime value assigned to this channel during creation.
+				Will be [code]65535[/code] if not specified.
 			</description>
 		</method>
 		<method name="get_max_retransmits" qualifiers="const">
 			<return type="int">
 			</return>
 			<description>
+				Returns the maxRetrismits value assigned to this channel during creation.
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCDataChannel.xml]: Variable typo: maxRetrismits => maxRetransmits

File: modules/webrtc/doc_classes/WebRTCDataChannel.xml [hunk 2]
```diff
@@ -11,85 +11,106 @@
...
 			</return>
 			<description>
+				Returns [code]true[/code] if this channel was created with ordering enabled (default).
 			</description>
 		</method>
 		<method name="poll">
 			<return type="int" enum="Error">
 			</return>
 			<description>
+				Reserved, but not used for now.
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCDataChannel.xml]: Side note for @godotengine/documentation : how do we manage documentation for reserved methods?

File: modules/webrtc/doc_classes/WebRTCMultiplayer.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			</return>
+			<argument index="0" name="peer" type="WebRTCPeerConnection">
+			</argument>
+			<argument index="1" name="peer_id" type="int">
+			</argument>
+			<argument index="2" name="unreliable_lifetime" type="int" default="1">
+			</argument>
+			<description>
+				Add a new peer to the mesh with the given [code]peer_id[/code]. The [WebRTCPeerConnection] must be in state [constant WebRTCPeerConnection.STATE_NEW].
+				Three channels will be created for reliable, unreliable, and ordered transport. The value of [code]unreliable_lifetime[/code] will be passed to the [code]"maxPacketLifetime"[/code] option when creating unreliable and ordered channels (see [method WebRTCPeerConnection.create_data_channel]).
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCMultiplayer.xml]: No need to double-quote maxPacketLifeTime here, it is already surrounded with [code][/code] :)

File: modules/webrtc/doc_classes/WebRTCMultiplayer.xml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+				Close all the add peer connections and channels, freeing all resources.
+			</description>
+		</method>
+		<method name="get_peer">
+			<return type="Dictionary">
+			</return>
+			<argument index="0" name="peer_id" type="int">
+			</argument>
+			<description>
+				Return a dictionary representation of the peer with given [code]peer_id[/code] with three keys. [code]"connection"[/code] containing the [WebRTCPeerConnection] to this peer, [code]"channels"[/code] an array of three [WebRTCDataChannel], and [code]"connected"[/code] a boolean representing if the peer connection is currently connected (all three channels are open).
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCMultiplayer.xml]: Same here, no need for double quotes in [code].

File: modules/webrtc/doc_classes/WebRTCPeerConnection.xml [hunk 5]
```diff
@@ -35,18 +46,38 @@
 }">
 			</argument>
 			<description>
+				Returns a new [WebRTCDataChannel] (or [code]null[/code] on failure) with given [code]label[/code] and optionally configured via the [code]options[/code] dictionary. This method can only be called when the connection is in state [constant STATE_NEW].
+				There are two ways to create a working data channel: either call [method create_data_channel] on only one of the peer and listen to [signal data_channel_received] on the other, or call [method create_data_channel] on both peers, with the same values, and the [code]"negotiated"[/code] option set to [code]true[/code].
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCPeerConnection.xml]: No need for double quotes in [code] (negotiated)

File: modules/webrtc/doc_classes/WebRTCPeerConnection.xml [hunk 6]
```diff
@@ -73,6 +121,8 @@
 			<argument index="1" name="sdp" type="String">
 			</argument>
 			<description>
+				Sets the SDP description of the local peer. This should be called in response to [signal session_description_created].
+				If [code]type[/code] is [code]"answer"[/code] the peer will start emitting [signal ice_candidate_created].
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCPeerConnection.xml]: No need for double quote in [code] (answer)

File: modules/webrtc/doc_classes/WebRTCPeerConnection.xml [hunk 7]
```diff
@@ -83,6 +133,9 @@
 			<argument index="1" name="sdp" type="String">
 			</argument>
 			<description>
+				Sets the SDP description of the remote peer. This should be called with the values generated by a remote peer and received over the signaling server.
+				If [code]type[/code] is [code]"offer"[/code] the peer will emit [signal session_description_created] with the appropriate answer.
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCPeerConnection.xml]: No need for double quote in [code], same in line 138

File: modules/webrtc/doc_classes/WebRTCPeerConnection.xml [hunk 8]
```diff
@@ -91,6 +144,8 @@
 			<argument index="0" name="channel" type="Object">
 			</argument>
 			<description>
+				Emitted when a new in-band channel is received, i.e. when the channel was created with [code]"negotiated": false[/code] (default).
```
Comment (StraToN) [inline modules/webrtc/doc_classes/WebRTCPeerConnection.xml]: No need for double quote in [code]

**Review by StraToN — CHANGES_REQUESTED** [review]
Just a few simple documentation typos and double quotes to remove :) Great job !

**Comment by Faless** [comment]
@StraToN thanks for the review! Updated with your suggestions :)

**Review by StraToN — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**